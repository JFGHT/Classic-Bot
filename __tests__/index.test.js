/* eslint-env jest */

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Router from 'next/router';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { middleware } from '../store/websocket.middleware';
import Index from '../pages/index';

// Mocking Router
const mockedRouter = { push: () => {}, prefetch: () => {} };
// Running the Router instance
Router.router = mockedRouter;

const middlewares = [thunk, middleware];
const initialState = {
  messages: [],
  events: [],
};
const initialProps = {
  chat: [],
  userEvents: [],
  botChat: [],
  connection: [],
  whispersArray: [],
};

describe('With Enzyme', () => {
  const mockStore = configureStore(middlewares);
  let store;
  let container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<Provider store={store}><Index /></Provider>);
  });

  it('Component is being rendered', () => {
    expect(container.length).toEqual(1);
  });

  it('Props as initialized', () => {
    Object.keys(initialProps).forEach(element => {
      expect(container.prop(element)).toEqual(expect.arrayContaining([]));
    });
  });
});

describe('With Snapshot Testing', () => {
  const mockStore = configureStore(middlewares);
  const store = mockStore(initialState);
  it('App starts with header and redux', () => {
    const component = renderer.create(<Provider store={store}><Index /></Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
