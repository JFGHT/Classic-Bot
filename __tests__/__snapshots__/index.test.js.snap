// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App starts with header and redux 1`] = `
<div>
  <div
    className="index__content container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="index__title__center__wrapper"
        >
          <img
            alt="Warcraft III Reforged"
            src="/static/reforged.png"
          />
        </div>
      </div>
    </div>
    <div
      className="index__title__row row"
    >
      <div
        className="col"
      >
        <div
          className="index__title__center__wrapper"
        >
          <div
            className="index__title__wrapper"
          >
            <h1
              className="index__title__title"
            >
              Classic Bot
            </h1>
            <small
              className="index__title__subtitle"
            >
              v1.0.0
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="index__title__center__wrapper"
        >
          <div
            className="position-relative form-group"
          >
            <input
              autoFocus={true}
              className="form-control"
              id="apiKey"
              name="apiKey"
              onChange={[Function]}
              placeholder="Enter your API key..."
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="index__button__row row"
    >
      <div
        className="col"
      >
        <div
          className="index__title__center__wrapper"
        >
          <button
            aria-label={null}
            className="btn btn-primary"
            onClick={[Function]}
            type="button"
          >
            Connect
          </button>
        </div>
      </div>
    </div>
    <div
      className="index__faq__row row"
    >
      <div
        className="col"
      >
        <div
          className="index__title__center__wrapper"
        >
          <a
            href="/about"
            onClick={[Function]}
          >
            What is Classic Bot?
          </a>
        </div>
      </div>
    </div>
    <div
      className="push"
    />
  </div>
  <footer
    className="index__footer"
  >
    <small>
      Warcraft® III: Reforged™ | ©2018 Blizzard Entertainment, Inc.. All rights reserved. This is not an official product.
    </small>
  </footer>
</div>
`;
